docs/handbook/03_frames.md

```diff
@@ -74,11 +74,11 @@ In the example above, the trays start empty, but it's also possible to populate
 
 ```html
 <turbo-frame id="set_aside_tray" src="/emails/set_aside">
-  <img src="/icons/spinner.gif">
+  <a href="/emails/set_aside">Set aside emails</a>
```

Comment:
How do you feel about declaring both, with the Set aside link wrapped in a `<noscript>` element?

Reply:
The spinner could be confusing if it is shown and the JavaScript does not run for whatever reason, and the user waits and thinks something is happening when it is not, so I think it would be best if the spinner was only shown if the relevant JavaScript is running. Similarly, JavaScript might be enabled (so `noscript` would not be shown) but something goes wrong loading/running the JavaScript, and then the user has no way to proceed, so I don't think wrapping it in `noscript` is the best way to proceed.

But yes, the spinner could be present initially in the HTML if you had, I dunno e.g.
```html
<img class="fetch-show" src="/icons/spinner.gif">
<a class="fetch-hide" href="/emails/set_aside">Set aside emails</a>
```
with some CSS like
```css
.fetch-show { display: none; }
```
and then the JS that kicks off the fetch / some event handler at that point / whatever shows `#set_aside_tray .fetch-show` and hides `#set_aside_tray .fetch-hide` – and then if something goes wrong fetching the page it can revert back to showing the link easily enough, which would also be a good thing.
Hope that's helpful.